src/js/controller/instream-html5.js

```diff
@@ -38,14 +44,20 @@ const InstreamHtml5 = function(_controller, _model) {
     };
 
     /** Load an instream item and initialize playback **/
-    _this.load = function(item) {
+    _this.load = function() {
+        // Let the player media model know we're using it's video tag
+        const mediaModelState = _model.mediaModel.attributes;
+        mediaModelState.setup = false;
+        mediaModelState.started = false;
 
-        _adModel.set('item', 0);
-        _adModel.set('playlistItem', item);
         // Make sure it chooses a provider
-        _adModel.setActiveItem(item);
-
-        // check provider after item change
```

Comment:
@cmulgannon @MaxMikheyenko @pajong 

The following changes will along with the changes in model.js should allow VAST to handle hls and mpd creative sources. The player will load the providers on the fly for creatives you pass it. We should look at the vast plugin creative filtering, and make sure it passes all video creatives as a prioritized playlist item source list the player can select from.